tests/test_auth.js

```diff
@@ -0,0 +1,96 @@
+const nock = require('nock');
+const assert = require('assert');
+const { morph } = require('mock-env');
+const { logger, BasicClient } = require('../lib');
+const { assertNoError } = require('./utils');
+
+
+const API_URL = 'http://fakeapi.foo/';
+
+logger.silent = true;
+
+
+describe('SmartFile Basic API client', () => {
+  it('can read config from env', (done) => {
+    /*
+    This test instantiates a client without any options.
+
+    It then ensures the client has picked up configuration options from the
+    environment. However, we use mock-env.morph to only temporarily set
+    "environment" variables for this test.
+    */
+
+    let client;
+
+    morph(() => {
+      client = new BasicClient();
+    }, {
+      SMARTFILE_API_URL: API_URL,
+      SMARTFILE_USER: 'foobar',
+      SMARTFILE_PASS: 'baz',
+    });
+
+    assert(client.baseUrl === API_URL);
+    assert(client.options.auth === 'foobar:baz');
+
+    done();
```

Comment:
Is there anything asynchronous here that requires `done()`?

Reply:
`done()` signals to mocha that the test has finished. Is there another way I am not aware of?

Reply:
I believe that it is only necessary to signal to mocha when something asynchronous has finished: https://mochajs.org/#asynchronous-code
Otherwise, you don't need the parameter at all and just let the function finish without an exception

Reply:
Wow, learn something every day. It must be doing some magic to check if the test function accepts the callback or not.